aw_core/config.py

```diff
@@ -1,12 +1,83 @@
 import os
 import logging
+from typing import Any, Dict, Union
 from configparser import ConfigParser
 
+from deprecation import deprecated
+import tomlkit
+
 from aw_core import dirs
+from aw_core.__about__ import __version__
 
 logger = logging.getLogger(__name__)
 
 
+def _merge(a: dict, b: dict, path=None):
+    """
+    Recursively merges b into a, with b taking precedence.
+
+    From: https://stackoverflow.com/a/7205107/965332
+    """
+    if path is None:
+        path = []
+    for key in b:
+        if key in a:
+            if isinstance(a[key], dict) and isinstance(b[key], dict):
+                _merge(a[key], b[key], path + [str(key)])
+            elif a[key] == b[key]:
+                pass  # same leaf value
+            else:
+                a[key] = b[key]
+        else:
+            a[key] = b[key]
+    return a
+
+
+def _comment_out_toml(s: str):
+    return "\n".join(["#" + line for line in s.split("\n")])
+
+
+def load_config_toml(
+    appname: str, default_config: str
+) -> Union[dict, tomlkit.container.Container]:
+    config_dir = dirs.get_config_dir(appname)
+    config_file_path = os.path.join(config_dir, "{}.toml".format(appname))
+
+    # Run early to ensure input is valid toml before writing
+    default_config_toml = tomlkit.parse(default_config)
+
+    # Override defaults from existing config file
+    if os.path.isfile(config_file_path):
+        with open(config_file_path, "r") as f:
+            config = f.read()
+        config_toml = tomlkit.parse(config)
+    else:
+        # TODO: If file doesn't exist, write with commented-out default config
+        with open(config_file_path, "w") as f:
+            f.write(_comment_out_toml(default_config))
+        config_toml = dict()
+
+    config = _merge(default_config_toml, config_toml)
+
+    return config
+
+
+def save_config_toml(appname: str, config: str) -> None:
+    # Check that passed config string is valid toml
+    assert tomlkit.parse(config)
```

Comment:
Should we have a schema to test against here and/or in the tests? 

Also does that assert result in a good runtime error?

Reply:
Since each user (server, watchers) of the API decides itself what config options/format to use I'm not sure if a schema makes sense. 

But maybe it could check against the default config or something to make sure things are of the same type etc? Sounds like a lot of work though, and it will probably work just fine and errors should be rare and clearly user-caused (since the user will know that they edited the config).

Reply:
Would be nice with a schema, but considering the work needed I don't think it's worth it.
I don't think the configs are large enough as of now that making a schema would add much value, if we find in the future that we would like that we can add that.